### PR TITLE
[ros2] Add default ign_gazebo version

### LINKED ISSUE #133: Launch file may start wrong ign-gazebo version
`ign_gazebo.launch` starts Ignition Gazebo with the `ign gazebo` command line command:

https://github.com/ignitionrobotics/ros_ign/blob/a4bc25fde117f9ce7279c83dc419bfb0eb3983b9/ros_ign_gazebo/launch/ign_gazebo.launch.py#L34

https://github.com/ignitionrobotics/ros_ign/blob/968c31586bbe9964a3788d5970a90f88dfc3cb25/ros_ign_gazebo/scripts/ign_gazebo#L16

The problem is that, if a user has more than one `ign-gazebo` version installed, that will launch the latest version, which may not necessarily be what the ROS user wants. For example, if I have `ros-foxy-ros-ign` (which uses Citadel, i.e. `ign-gazebo3`) and Ignition Dome (i.e. `ign-gazebo4`) installed, the launch file will start a version of Ignition that may not be compatible with the rest of the things being launched.

The command line tool accepts the `--force-version` argument to specify what version to launch. But that is quite strict at the moment, and requires passing the fully qualified version, including minor and patch numbers (see https://github.com/ignitionrobotics/ign-tools/issues/36). This makes it complicated to hardcode the version on the launch file.

### PR DESCRIPTION
**Pull request by caioaamaral**
# 🦟 Bug fix

Fixes #133 

## Summary
This change uses as the default version the dependency resolved by the package manifest (which uses ignition-gazebo3 for the foxy distro). This is accomplished by appending the flag `--force-version` (only when it's not already set) to the `ign_args`

## Checklist
- [x] Signed all commits for DCO
- [ ] Added tests
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

- [x] **Depends on**: ignitionrobotics/ign-tools#44 (to allow passing the major version only)

**Note to maintainers**: Remember to use **Squash-Merge**

### CODE REVIEW

**Comment by caioaamaral** [comment]
I added to the checklist another pr which this one depends on (ignitionrobotics/ign-tools#44)

Is a unit test desired? (maybe we could test it by using specific configurations with the ci?)

**Comment by caioaamaral** [comment]
@chapulina I've opened the PR to the wrong branch, could you please move it to the ros2 branch please? :facepalm: 

edit: fixed

File: ros_ign_gazebo/launch/ign_gazebo.launch.py [hunk 1]
```diff
@@ -14,28 +14,51 @@
...
+
+
+def __get_ign_args(ign_args):
+    if '--force_version' not in ign_args:
+        ign_args += '--force_version {}'.format(__IGN_GAZEBO_DEFAULT_VERSION)
+    
+    return ign_args
+
+
+def __get_ign_gazebo_default_version(): # this will fail if the package build was performed with '--merge-install'
```
Comment (caioaamaral) [inline ros_ign_gazebo/launch/ign_gazebo.launch.py]: To the reviewer:

At least locally I couldn't find the package when my workspace was build with `--merge-install` flag, apparently the overlay does not work with such configuration (also noticed that the package manifest was not installed).

**Review by ahcorde — CHANGES_REQUESTED** [review]
do you mind to fix the conflicts with the main branch ?

**Comment by caioaamaral** [comment]
> do you mind to fix the conflicts with the main branch ?

Done

**Comment by caioaamaral** [comment]
Friendly ping.

I think this one is good to go now that ignitionrobotics/ign-tools#44 is merged

**Comment by chapulina** [comment]
Thanks, @caioaamaral , we'll take a look soon. Note that we'll need an `ign-tools` release, and also upload the released version to packages.ros.org, before this can be merged.

**Comment by chapulina** [comment]
I'm getting an error on launch, I tried compiling with `--merge-install` and without. Do you know what's happening?

```
  File "/home/chapulina/dev_focal/ws_foxy/install/ros_ign_gazebo/share/ros_ign_gazebo/launch/ign_gazebo.launch.py", line 39, in generate_lau
nch_description
    OpaqueFunction(__get_ign_args)
TypeError: __init__() takes 1 positional argument but 2 were given
```

**Comment by chapulina** [comment]
@caioaamaral , are you still working on this PR?

**Comment by caioaamaral** [comment]
> @caioaamaral , are you still working on this PR?

Sorry for the delay. I'll have a look by the end of this week

**Comment by chapulina** [comment]
Hi @caioaamaral , I'm closing this due to inactivity, but feel free to reopen if you start working on it again. Thanks!

**Pull request closed without merging**